src/transformers/trainer.py

```diff
@@ -370,6 +370,19 @@ def __init__(
         else:
             self.is_model_parallel = False
 
+        if (
+            getattr(model, "hf_device_map", None) is not None
+            and len(set(model.hf_device_map.values())) > 1
+            and not self.is_model_parallel
+        ):
+            self.is_model_parallel = True
+
+            # warn users
+            logger.warning(
```

Comment:
An info would be enough here.

Reply:
Fixed in [5eb72b4](https://github.com/huggingface/transformers/pull/22532/commits/5eb72b4b8f0d80b8de3ff11c867ff7207fde6bd8)